core/src/main/java/hudson/model/queue/MappingWorksheet.java

```diff
@@ -370,7 +370,7 @@ public MappingWorksheet(BuildableItem item, List<? extends ExecutorSlot> offers,
             if (ec.node==null)  continue;   // evict out of sync node
             executors.add(ec);
         }
-        this.executors = ImmutableList.copyOf(executors);
+        this.executors = Collections.unmodifiableList(executors);
```

Comment:
The local variable `executors` was declared within this method and goes out of scope at the end of the method, so nothing else can modify it. We do not need to copy it.